x/distribution/keeper/querier.go

```diff
@@ -314,3 +318,11 @@ func queryDelegatorWithdrawAddress(ctx sdk.Context, _ []string, req abci.Request
 
 	return bz, nil
 }
+
+func queryCommunityPool(ctx sdk.Context, _ []string, req abci.RequestQuery, k Keeper) ([]byte, sdk.Error) {
+	bz, err := k.cdc.MarshalJSON(k.GetFeePoolCommunityCoins(ctx))
+	if err != nil {
+		return nil, sdk.ErrInternal(sdk.AppendMsgToErr("could not marshal result to JSON", err.Error()))
```

Comment:
When would this ever fail should this not be a panic? (aka use MustMarshalJSON)

Reply:
Should it not return an error? I don't think the node should crash/panic here when it fails to unmarshal, but rather bubble up the error to the caller.

Reply:
well if it fails to unmarshal it means the state is somehow corrupted correct? which means it would be appropriate for the node to crash

Reply:
I don't think it's okay for a node to crash from a querier.

Reply:
This function returns an error. IMHO it should be caller's responsibility to handle the error.

Reply:
olright